src/job/jobHandler.ts

```diff
@@ -234,21 +234,90 @@ export abstract class JobHandler {
     }
   }
 
+  private async loggingMessage(resp: CommandExecutorResponse): Promise<void> {
+    await this._logger.save(
+      this.currJob._id,
+      `${'(BUILD)'.padEnd(15)}worker.sh run details:\n\n${resp.output}\n---\n${resp.error}`
+    );
+    if (resp.status != 'success') {
+      const error = new AutoBuilderError(resp.error, 'BuildError');
+      await this.logError(error);
+      throw error;
+    }
+  }
+
+  // call this method when we want benchmarks and uses cwd option to call command outside of a one liner.
+  private async callWithBenchmark(command: string, stage: string): Promise<CommandExecutorResponse> {
+    const start = performance.now();
+    const resp = await this._commandExecutor.execute([command], `repos/${this.currJob.payload.repoName}`);
+    await this._logger.save(
+      this.currJob._id,
+      `${'(COMMAND)'.padEnd(15)} ${command} run details in ${this.currJob.payload.repoName}`
+    );
+    const end = performance.now();
+    const update = {
+      [`${stage}StartTime`]: start,
+      [`${stage}EndTime`]: end,
+    };
+    await this._jobRepository.findOneAndUpdateExecutionTime(this.currJob._id, update);
+    return resp;
+  }
+
+  private async exeBuildModified(): Promise<void> {
+    const stages = {
+      ['get-build-dependencies']: 'buildDepsExe',
+      ['next-gen-parse']: 'parseExe',
+      ['next-gen-html']: 'htmlExe',
+      ['oas-page-build']: 'oasPageBuildExe',
+    };
+
+    // get the prerequisite commands which should be all commands up to `rm -f makefile`
+    const endOfPrerequisiteCommands = this.currJob.buildCommands.indexOf('rm -f makefile');
+    const index = endOfPrerequisiteCommands + 1;
+    const prerequisiteCommands = this.currJob.buildCommands.slice(0, index);
+    await this._logger.save(this.currJob._id, `'(PREREQUISITE COMMANDS)'${prerequisiteCommands.join(' && ')}`);
+
+    const makeCommands = this.currJob.buildCommands.slice(index);
+    await this._logger.save(this.currJob._id, `'(MAKE COMMANDS)'${makeCommands.join(' && ')}`);
+
+    // call prerequisite commands
+    const prerequisiteResp = await this._commandExecutor.execute(prerequisiteCommands);
+    await this.loggingMessage(prerequisiteResp);
```

Comment:
is this required? seems like these are
```
      `. /venv/bin/activate`,
      `cd repos/${this.currJob.payload.repoName}`,
      `rm -f makefile`,
```
and not sure if we want to log this as well as including them with the log message `worker.sh run details: ` as in `loggingMessage`

Reply:
@seungpark I am not sure what you are asking here, sorry.

As for the logging, I guess we don't need to log those as there will not be an output. I can no longer pass the response to the logger message method.

Reply:
the `prerequisiteResp` are the ones listed above, and i suspect this will be logged as 

```
"(BUILD)        worker.sh run details:

<result of cd and rm>        <------------------- is this relevant?
---
"(BUILD)        worker.sh run details:

<result of make commands>

```

Reply:
@seungpark that is fair, probably no value in logging those. I will not pass the response from the prerequisite commands. Good callout.